### PR TITLE
Additional Custom Control Methods

### PR DESCRIPTION
**Pull request by TurtIeSocks**
Motivation is for React reasons, since an error is thrown if a button has already been mounted (when the component rerenders...), these checks are helpful to avoid issues.

- adds `buttonExists(name: string)` to get the status of a button
- adds `getCustomButtons()` that returns an object of all of the custom controls
- ts definitions

### CODE REVIEW

File: src/js/Toolbar/L.PM.Toolbar.js [hunk 1]
```diff
@@ -563,7 +563,16 @@ const Toolbar = L.Class.extend({
...
+  controlExists(name) {
+    return Boolean(this.getButtons()[name]);
+  },
+  getCustomControls() {
+    return Object.fromEntries(
+      Object.entries(this.getButtons()).filter(
+        ([, v]) => v._button.tool === 'custom'
+      )
+    );
+  },
```
Comment (Falke-Design) [inline src/js/Toolbar/L.PM.Toolbar.js:575]: Let us make this more general:

```suggestion
getButtonsInBlock(name) {
  const buttonsInBlock = {};
    if(name){
    for(const buttonName in map.pm.Toolbar.getButtons()){
      const button = map.pm.Toolbar.getButtons()[buttonName];
      // draw controls doesn't have a block
      if(button._button.tool === name || (name === 'draw' && !button._button.tool)){
        buttonsInBlock[buttonName] = button;
      }
    }
  }
  return buttonsInBlock;
}
```

File: src/js/Toolbar/L.PM.Toolbar.js [hunk 2]
```diff
@@ -563,7 +563,16 @@ const Toolbar = L.Class.extend({
     this.changeControlOrder();
     return control;
   },
-
+  controlExists(name) {
+    return Boolean(this.getButtons()[name]);
+  },
```
Comment (Falke-Design) [inline src/js/Toolbar/L.PM.Toolbar.js:568]: Would it be ok for you to implementend `getButton(name)`? It returns `undefined` if it doesn't exists.

File: README.md [hunk 3]
```diff
@@ -1056,6 +1056,9 @@ The following methods are available on `map.pm.Toolbar`:
 | createCustomControl(`options`)            | -        | To add a custom Control to the Toolbar. [Details](#adding-newcustom-controls)                                                                                                                    |
 | copyDrawControl(`instance`, `options`)    | `Object` | Creates a copy of a draw Control. Returns the `drawInstance` and the `control`.                                                                                                                  |
 | changeActionsOfControl(`name`, `actions`) | -        | Change the actions of an existing button.                                                                                                                                                        |
+| getButtons() | `Object`       | Get an object of the current buttons button.                                                                                                                                                        |
```
Comment (Falke-Design) [inline README.md:1059]: 👍

**Review by Falke-Design — CHANGES_REQUESTED** [review]
Please check my review. 

If you want, it would be helpful to have following methods too:
- L.Controls: `disabled() {return  !!this._button.disabled}`
- L.Controls: `visible(){return !!this._map}`
- L.Controls: `getButtonsOptions() {return this._button}`

**Review by Falke-Design — CHANGES_REQUESTED** [review]
And please add tests for the new methods

**Comment by TurtIeSocks** [comment]
Apologies for the delay @Falke-Design, finally came back to this and fixed suggestions.

File: src/js/Toolbar/L.PM.Toolbar.js [hunk 4]
```diff
@@ -563,7 +563,28 @@ const Toolbar = L.Class.extend({
     this.changeControlOrder();
     return control;
   },
-
+  controlExists(name) {
+    return Boolean(this.getButtons()[name]);
```
Comment (Falke-Design) [inline src/js/Toolbar/L.PM.Toolbar.js:567]: let us change this to:
```suggestion
    return Boolean(this.getButton(name));
```

File: src/js/Toolbar/L.PM.Toolbar.js [hunk 5]
```diff
@@ -563,7 +563,28 @@ const Toolbar = L.Class.extend({
...
+  },
+  getButton(name) {
+    return this.getButtons()[name];
+  },
+  getButtonsInBlock(name) {
+    const buttonsInBlock = {};
+    if (name) {
+      for (const buttonName in this.getButtons()) {
+        const button = this.getButtons()[buttonName];
```
Comment (Falke-Design) [inline src/js/Toolbar/L.PM.Toolbar.js:576]:
```suggestion
        const button = this.getButton(buttonName);
```

**Review by Falke-Design — CHANGES_REQUESTED** [review]

File: src/js/Toolbar/L.PM.Toolbar.js [hunk 6]
```diff
@@ -563,7 +563,28 @@ const Toolbar = L.Class.extend({
...
+    return Boolean(this.getButton(name));
+  },
+  getButton(name) {
+    return this.getButtons()[name];
+  },
+  getButtonsInBlock(name) {
+    const buttonsInBlock = {};
+    if (name) {
+      for (const buttonName in this.getButtons()) {
+        const button = this.getButton(buttonName);
```
Comment (Falke-Design) [inline src/js/Toolbar/L.PM.Toolbar.js:576]:
```suggestion
        const button = this.getButtons()[buttonName];
```

**Comment by Falke-Design** [comment]
Thank you for your contribution!

**Pull request merged**